include/userobjects/PKAGeneratorAlphaDecay.h

```diff
@@ -0,0 +1,52 @@
+/**********************************************************************/
+/*                     DO NOT MODIFY THIS HEADER                      */
+/* MAGPIE - Mesoscale Atomistic Glue Program for Integrated Execution */
+/*                                                                    */
+/*            Copyright 2017 Battelle Energy Alliance, LLC            */
+/*                        ALL RIGHTS RESERVED                         */
+/**********************************************************************/
+
+#ifndef PKAGENERATORALPHADECAY_H
+#define PKAGENERATORALPHADECAY_H
+
+#include "PKAGeneratorBase.h"
+
+class PKAGeneratorAlphaDecay;
+
+template<>
+InputParameters validParams<PKAGeneratorAlphaDecay>();
+
+/**
+ * PKAGenerator for alpha decays and corresponding recoil atoms
+ */
+class PKAGeneratorAlphaDecay : public PKAGeneratorBase
+{
+public:
+  PKAGeneratorAlphaDecay(const InputParameters & parameters);
+
+  virtual void appendPKAs(std::vector<MyTRIM_NS::IonBase> & ion_list, Real dt, Real vol, const MyTRIMRasterizer::AveragedData &) const override;
+
+  /// this stores the decay information for a single alpha decaying nuclide
+  struct DecayData {
+    DecayData(unsigned int n = 0) : _n_decay_channels(n), _decay_constants(n, 0.0), _alpha_energies(n, 0.0), _intensities(n, 0) {}
+
+    unsigned int _n_decay_channels;
+    std::vector<Real> _decay_constants;
+    std::vector<Real> _alpha_energies;
+    std::vector<Real> _intensities;
+  };
```

Comment:
I would make this a vector of structs rather than a struct of vectors. I.e. create a `DecayChannel` struct with constant, energy, intensity scalar members. That way you cannot have inconsistent vector length and resizing is a simpler operation.